src/cpp/src/speculative_decoding/speculative_decoding_stateful_eagle3.hpp

```diff
@@ -0,0 +1,297 @@
+        if (m_sequence_group) {
+            auto sequences = m_sequence_group->get_running_sequences();
+            if (index < sequences.size()) {
+                return sequences[index];
+            }
+        }
+        return nullptr;
+    }
+
+    /// @brief Returns current (first) sequence - convenience method for top-1
+    /// @deprecated Use get_sequence(0) for clarity, will be removed when top-k is implemented
+    Sequence::Ptr get_current_sequence() const {
+        return get_sequence(0);
+    }
+
+    ov::Tensor get_logits() const;
+    ov::Tensor get_hidden_features() const;
+
+    void build_model_inputs(size_t token_count,
+                            ov::Tensor& input_ids,
+                            ov::Tensor& attention_mask,
+                            ov::Tensor& position_ids);
+
+    ov::Tensor create_hidden_state_placeholder(const ov::Shape& shape) const;
+
+    /// @brief Samples tokens from logits
+    /// @param num_tokens_to_validate Draft tokens to validate (0 for standard sampling)
+    /// @return Single token or vector of validated tokens
+    std::variant<int64_t, std::vector<int64_t>> sample_tokens(const ov::Tensor& logits,
+                                                              size_t input_token_count,
+                                                              size_t sample_count,
+                                                              size_t num_tokens_to_validate = 0);
+
+    /// @brief Executes forward pass: prepare inputs, infer, and sample
+    virtual InferResult forward(const InferContext& ctx) = 0;
+
+    ov::genai::RawPerfMetrics& get_raw_perf_metrics() {
+        return m_raw_perf_metrics;
+    }
+
+protected:
+    static constexpr size_t BATCH_SIZE = 1;
+
+    uint64_t execute_inference();
+    void update_performance_metrics(uint64_t inference_time_us, size_t tokens_count);
+
+    void log_debug(LogCategory category, const std::string& message) const;
+    void log_tensor_info(const std::string& name, const ov::Tensor& tensor) const;
+    void log_tensor_content(const std::string& name, const ov::Tensor& tensor, size_t max_elements = 10) const;
+    void log_model_inputs(const ov::Tensor& input_ids,
+                          const ov::Tensor& attention_mask,
+                          const ov::Tensor& position_ids) const;
+    void log_model_outputs(const ov::Tensor& logits, const ov::Tensor& hidden_features) const;
+
+    std::string m_device;
+    ov::AnyMap m_properties;
+    ov::genai::Tokenizer m_tokenizer;
+    mutable ov::InferRequest m_request;
+    ov::genai::utils::KVAxesPosition m_kv_axes_pos;
+    size_t m_max_prompt_len = 0;
+
+    SequenceGroup::Ptr m_sequence_group;
+    Sampler m_sampler;
+    ov::genai::RawPerfMetrics m_raw_perf_metrics;
+    bool m_verbose = false;
+};
+
+/**
+ * @brief Target model wrapper for Eagle3
+ *
+ * Validates draft predictions and generates final output
+ */
+class Eagle3TargetModelWrapper : public Eagle3InferWrapperBase {
```

Comment:
Usually, in OpenVINO, we first read model, then compile model, then create infer_request out of last. But here we name specific to the target inference wrapper with the `model` word inside. It confuses me a bit. 
Would `Eagle3TargetWrapper` work for you?